sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UriShouldNotBeHardcoded.cs

```diff
@@ -0,0 +1,156 @@
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    if (c.IsTest())
+                    {
+                        return;
+                    }
+
+                    var stringLiteral = (LiteralExpressionSyntax)c.Node;
+                    if (IsInCheckedContext(stringLiteral, c.SemanticModel) &&
+                        UriRegex.IsMatch(stringLiteral.Token.ValueText))
+                    {
+                        c.ReportDiagnostic(Diagnostic.Create(rule, stringLiteral.GetLocation(), AbsoluteUriMessage));
+                    }
+                }, SyntaxKind.StringLiteralExpression);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    if (c.IsTest())
+                    {
+                        return;
+                    }
+
+                    var addExpression = (BinaryExpressionSyntax)c.Node;
+                    if (!IsInCheckedContext(addExpression, c.SemanticModel))
+                    {
+                        return;
+                    }
+
+                    if (IsPathDelimiter(addExpression.Left))
+                    {
+                        c.ReportDiagnostic(Diagnostic.Create(rule, addExpression.Left.GetLocation(),
+                            PathDelimiterMessage));
+                    }
+
+                    if (IsPathDelimiter(addExpression.Right))
+                    {
+                        c.ReportDiagnostic(Diagnostic.Create(rule, addExpression.Right.GetLocation(),
+                            PathDelimiterMessage));
+                    }
+                }, SyntaxKind.AddExpression);
+        }
+
+        private static bool IsInCheckedContext(ExpressionSyntax expression, SemanticModel model)
+        {
+            var argument = expression.FirstAncestorOrSelf<ArgumentSyntax>();
+            if (argument != null)
+            {
+                var argumentIndex = (argument.Parent as ArgumentListSyntax)?.Arguments.IndexOf(argument);
+                if (argumentIndex == null ||
+                    argumentIndex < 0)
+                {
+                    return false;
+                }
+
+                var constructorOrMethod = argument.Ancestors()
+                    .FirstOrDefault(ancestor => ancestor.IsKind(SyntaxKind.InvocationExpression) ||
+                                                ancestor.IsKind(SyntaxKind.ObjectCreationExpression));
+                var methodSymbol = constructorOrMethod != null
+                    ? model.GetSymbolInfo(constructorOrMethod).Symbol as IMethodSymbol
+                    : null;
+
+                return (methodSymbol != null && argumentIndex.Value < methodSymbol.Parameters.Length)
+                    ? methodSymbol.Parameters[argumentIndex.Value].Name.SplitCamelCaseToWords()
+                        .Any(name => checkedVariableNames.Contains(name))
+                    : false;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the unnecessary Boolean literal(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1125)